src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -391,28 +391,25 @@ private void Poll()
 
             try
             {
-                if (_requests.IsEmpty())
-                {
-                    // no outstanding requests so we don't expect any messages back, but we should anyway
-                    // drive the KafkaConsumer by polling to handle partition events etc.
-                    PausePartitions(currentAssignment);
-                    var message = _consumer.Consume(TimeSpan.FromMilliseconds(1));
-                    if (message != null)
-                    {
-                        throw new IllegalActorStateException($"Got unexpected Kafka message: {message.ToJson()}");
-                    }
-                }
-                else
+                // resume partitions to fetch
+                IImmutableSet<TopicPartition> partitionsToFetch = _requests.Values.SelectMany(v => v.Topics).ToImmutableHashSet();
+                var resumeThese = currentAssignment.Where(partitionsToFetch.Contains).ToList();
+                var pauseThese = currentAssignment.Except(resumeThese).ToList();
+                PausePartitions(pauseThese);
+                ResumePartitions(resumeThese);
+
+                ConsumeResult<K, V> consumed; 
+                var pooled = new List<ConsumeResult<K, V>>();
+                var deadline = _settings.PollTimeout.TotalMilliseconds;
+                var watch = Stopwatch.StartNew();
+                do
```

Comment:
Instead of consuming one message at a time within a timeout limit, we poll all available messages within that timeout limit. This is actually close to JVM implementation behaviour.